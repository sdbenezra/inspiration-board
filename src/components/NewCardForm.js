import React, { Component } from 'react';
import PropTypes from 'prop-types';
// import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  static propTypes = {
    addCardCallback: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);

    this.state = {
      cards:[{
        cardID: {
          card: {text: '', emoji: '',}
        }
      }]
    }
  }

  onInputChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;

    const newState = {};
    newState[field] = value;
    this.setState(newState)
  }

  resetState = () => {
    this.setState({
      cards:[{cardID: {card: {text: '', emoji: '',}}}]
    });
  }


  onFormSubmit = (event) => {
    event.preventDefault();

    const card = {
      text: this.state.message,
      emoji: this.state.emoji,
    }
    this.resetState();
    this.props.addCardCallback(card)
  }

  render() {
    const emojiList =  EMOJI_LIST.map((item, i) => {
        return <option key={i} value={item}>{item}</option>
      });

    return(
      <form onSubmit={this.onFormSubmit} className="new-card-form" >
        <header className="new-card-form__header">
          Add a New Card!
        </header>
        <p></p>
        <section>
          <label htmlFor="message" className="new-card-form__form-label">Your Message</label>
          <input name="message" type="text" value={this.state.message} onChange={this.onInputChange} className="new-card-form__form-textarea"/>
            <select onChange={this.onInputChange} name="emoji" className="new-card-form__form-select">
              {emojiList}
            </select>
        </section>
        <p></p>
        <input type="submit" value="Add message" className="new-card-form__form-button" />
      </form>
    );
  }
}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func,
};

export default NewCardForm;
