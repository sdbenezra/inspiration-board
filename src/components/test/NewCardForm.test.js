import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';

describe('New Card Form', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow(
       <NewCardForm
          message = 'message'
          emoji = 'emoji'
          />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
