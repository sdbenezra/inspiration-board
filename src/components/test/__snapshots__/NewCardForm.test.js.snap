// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Card Form that it matches an existing snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    emoji="emoji"
    message="message"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="new-card-form__header"
>
          Add a New Card!
</header>,
        <p />,
        <section>
          <label
                    className="new-card-form__form-label"
                    htmlFor="message"
          >
                    Your Message
          </label>
          <input
                    className="new-card-form__form-textarea"
                    name="message"
                    onChange={[Function]}
                    type="text"
                    value={undefined}
          />
          <select
                    className="new-card-form__form-select"
                    name="emoji"
                    onChange={[Function]}
          >
                    <option
                              value=""
                    />
                    <option
                              value="heart_eyes"
                    >
                              heart_eyes
                    </option>
                    <option
                              value="beer"
                    >
                              beer
                    </option>
                    <option
                              value="clap"
                    >
                              clap
                    </option>
                    <option
                              value="sparkling_heart"
                    >
                              sparkling_heart
                    </option>
                    <option
                              value="heart_eyes_cat"
                    >
                              heart_eyes_cat
                    </option>
                    <option
                              value="dog"
                    >
                              dog
                    </option>
          </select>
</section>,
        <p />,
        <input
          className="new-card-form__form-button"
          type="submit"
          value="Add message"
/>,
      ],
      "className": "new-card-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a New Card!",
          "className": "new-card-form__header",
        },
        "ref": null,
        "rendered": "Add a New Card!",
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="new-card-form__form-label"
              htmlFor="message"
>
              Your Message
</label>,
            <input
              className="new-card-form__form-textarea"
              name="message"
              onChange={[Function]}
              type="text"
              value={undefined}
/>,
            <select
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
>
              <option
                            value=""
              />
              <option
                            value="heart_eyes"
              >
                            heart_eyes
              </option>
              <option
                            value="beer"
              >
                            beer
              </option>
              <option
                            value="clap"
              >
                            clap
              </option>
              <option
                            value="sparkling_heart"
              >
                            sparkling_heart
              </option>
              <option
                            value="heart_eyes_cat"
              >
                            heart_eyes_cat
              </option>
              <option
                            value="dog"
              >
                            dog
              </option>
</select>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Message",
              "className": "new-card-form__form-label",
              "htmlFor": "message",
            },
            "ref": null,
            "rendered": "Your Message",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-textarea",
              "name": "message",
              "onChange": [Function],
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value=""
/>,
                <option
                  value="heart_eyes"
>
                  heart_eyes
</option>,
                <option
                  value="beer"
>
                  beer
</option>,
                <option
                  value="clap"
>
                  clap
</option>,
                <option
                  value="sparkling_heart"
>
                  sparkling_heart
</option>,
                <option
                  value="heart_eyes_cat"
>
                  heart_eyes_cat
</option>,
                <option
                  value="dog"
>
                  dog
</option>,
              ],
              "className": "new-card-form__form-select",
              "name": "emoji",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "value": "",
                },
                "ref": null,
                "rendered": "",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "heart_eyes",
                  "value": "heart_eyes",
                },
                "ref": null,
                "rendered": "heart_eyes",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "beer",
                  "value": "beer",
                },
                "ref": null,
                "rendered": "beer",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": "clap",
                  "value": "clap",
                },
                "ref": null,
                "rendered": "clap",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "host",
                "props": Object {
                  "children": "sparkling_heart",
                  "value": "sparkling_heart",
                },
                "ref": null,
                "rendered": "sparkling_heart",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "host",
                "props": Object {
                  "children": "heart_eyes_cat",
                  "value": "heart_eyes_cat",
                },
                "ref": null,
                "rendered": "heart_eyes_cat",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "host",
                "props": Object {
                  "children": "dog",
                  "value": "dog",
                },
                "ref": null,
                "rendered": "dog",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "new-card-form__form-button",
          "type": "submit",
          "value": "Add message",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="new-card-form__header"
>
            Add a New Card!
</header>,
          <p />,
          <section>
            <label
                        className="new-card-form__form-label"
                        htmlFor="message"
            >
                        Your Message
            </label>
            <input
                        className="new-card-form__form-textarea"
                        name="message"
                        onChange={[Function]}
                        type="text"
                        value={undefined}
            />
            <select
                        className="new-card-form__form-select"
                        name="emoji"
                        onChange={[Function]}
            >
                        <option
                                    value=""
                        />
                        <option
                                    value="heart_eyes"
                        >
                                    heart_eyes
                        </option>
                        <option
                                    value="beer"
                        >
                                    beer
                        </option>
                        <option
                                    value="clap"
                        >
                                    clap
                        </option>
                        <option
                                    value="sparkling_heart"
                        >
                                    sparkling_heart
                        </option>
                        <option
                                    value="heart_eyes_cat"
                        >
                                    heart_eyes_cat
                        </option>
                        <option
                                    value="dog"
                        >
                                    dog
                        </option>
            </select>
</section>,
          <p />,
          <input
            className="new-card-form__form-button"
            type="submit"
            value="Add message"
/>,
        ],
        "className": "new-card-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a New Card!",
            "className": "new-card-form__header",
          },
          "ref": null,
          "rendered": "Add a New Card!",
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="new-card-form__form-label"
                htmlFor="message"
>
                Your Message
</label>,
              <input
                className="new-card-form__form-textarea"
                name="message"
                onChange={[Function]}
                type="text"
                value={undefined}
/>,
              <select
                className="new-card-form__form-select"
                name="emoji"
                onChange={[Function]}
>
                <option
                                value=""
                />
                <option
                                value="heart_eyes"
                >
                                heart_eyes
                </option>
                <option
                                value="beer"
                >
                                beer
                </option>
                <option
                                value="clap"
                >
                                clap
                </option>
                <option
                                value="sparkling_heart"
                >
                                sparkling_heart
                </option>
                <option
                                value="heart_eyes_cat"
                >
                                heart_eyes_cat
                </option>
                <option
                                value="dog"
                >
                                dog
                </option>
</select>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Message",
                "className": "new-card-form__form-label",
                "htmlFor": "message",
              },
              "ref": null,
              "rendered": "Your Message",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-textarea",
                "name": "message",
                "onChange": [Function],
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value=""
/>,
                  <option
                    value="heart_eyes"
>
                    heart_eyes
</option>,
                  <option
                    value="beer"
>
                    beer
</option>,
                  <option
                    value="clap"
>
                    clap
</option>,
                  <option
                    value="sparkling_heart"
>
                    sparkling_heart
</option>,
                  <option
                    value="heart_eyes_cat"
>
                    heart_eyes_cat
</option>,
                  <option
                    value="dog"
>
                    dog
</option>,
                ],
                "className": "new-card-form__form-select",
                "name": "emoji",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "heart_eyes",
                    "value": "heart_eyes",
                  },
                  "ref": null,
                  "rendered": "heart_eyes",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "beer",
                    "value": "beer",
                  },
                  "ref": null,
                  "rendered": "beer",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": "clap",
                    "value": "clap",
                  },
                  "ref": null,
                  "rendered": "clap",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "host",
                  "props": Object {
                    "children": "sparkling_heart",
                    "value": "sparkling_heart",
                  },
                  "ref": null,
                  "rendered": "sparkling_heart",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "host",
                  "props": Object {
                    "children": "heart_eyes_cat",
                    "value": "heart_eyes_cat",
                  },
                  "ref": null,
                  "rendered": "heart_eyes_cat",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "6",
                  "nodeType": "host",
                  "props": Object {
                    "children": "dog",
                    "value": "dog",
                  },
                  "ref": null,
                  "rendered": "dog",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-card-form__form-button",
            "type": "submit",
            "value": "Add message",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
